trtools/dumpSTR/filters.py

```diff
@@ -440,21 +488,42 @@ class HipSTRCallMinSuppReads(Reason):
 
     def __init__(self, threshold):
         self.threshold = threshold
-    def __call__(self, sample):
-        if sample["ALLREADS"] is None: return 0
+    def __call__(self, record: trh.TRRecord):
+        called_samples = record.GetCalledSamples()
+        if not np.any(called_samples):
+            return np.full((record.GetNumSamples()), np.nan)
+
+        if "ALLREADS" not in record.format:
+            sample_filter = np.zeros((record.GetNumSamples()), dtype=float)
+            return sample_filter
+        called_samples = np.logical_and(
+            called_samples, record.format["ALLREADS"] != '.'
+        )
+
         delim = "|"
-        if "/" in sample["GB"]: delim = "/"
-        gb = [int(item) for item in sample["GB"].split(delim)]
-        allreads = sample["ALLREADS"].split(";")
-        r1 = 0
-        r2 = 0
-        for item in allreads:
-            allele, readcount = [int(item) for item in item.split("|")]
-            if allele == gb[0]: r1 += readcount
-            if allele == gb[1]: r2 += readcount
-        min_read_count = min([r1, r2])
-        if min_read_count < self.threshold: return min_read_count
-        else: return None
+        # Going to assume that either all samples are phased or none are
```

Comment:
I don't think this assumption necessarily holds. Not sure though, you have worked with HipSTR data more than I.

Reply:
Not sure if this is a reasonable assumption, I agree its a bit sketchy. Its what was already being done though, so I'm viewing it as an issue that isn't being fixed which I'm okay with as opposed to an issue I'm introducing.

Reply:
Honestly, none of our code is sensitive to phasing. I'm not sure where else that is an issue, but it isn't being tested for at all. Handling indeterminate phasing properly would probably involve a lot of testing in a lot of places.